### PR TITLE
fix api build & local dev

### PR DESCRIPTION
**Pull request by b4l**
This will fix the api build.

### CODE REVIEW

File: api/Dockerfile [hunk 1]
```diff
@@ -16,7 +16,6 @@ WORKDIR /app
 COPY Cargo.toml Cargo.lock ./
 RUN mkdir src; echo "fn main() {}" > src/main.rs
 RUN cargo fetch --target x86_64-unknown-linux-musl
-RUN cargo build --offline --target x86_64-unknown-linux-musl --release
```
Comment (gberaudo) [inline api/Dockerfile:19]: Why this change?

File: api/DockerfileDev [hunk 2]
```diff
@@ -15,7 +15,7 @@ WORKDIR /app
 # We need a stub entry point for that
 COPY Cargo.toml Cargo.lock ./
 RUN mkdir src; echo "fn main() {}" > src/main.rs
-RUN cargo build --target x86_64-unknown-linux-musl
```
Comment (gberaudo) [inline api/DockerfileDev:18]: Why this change?

**Review by gberaudo — CHANGES_REQUESTED** [review]
These 2 changes break image build optimization. Let's discuss this before merging.

File: api/DockerfileDev (hunk 2 above)
Comment (b4l) [reply api/DockerfileDev:18]: this didn't seem to break caching locally. is https://github.com/swissgeol/ngm/pull/834/commits/fe73733a95a7edf989d884dff798eb9ef19af84d working for you?

File: api/Dockerfile (hunk 1 above)
Comment (b4l) [reply api/Dockerfile:19]: we build later again and i am not sure about the caching, don't care performance for a release but since we run it all the time https://github.com/swissgeol/ngm/pull/834/commits/fe73733a95a7edf989d884dff798eb9ef19af84d might be better.

**Review by gberaudo — APPROVED** [review]
Could you please squash and merge?
